lib/matplotlib/tests/test_figure.py

```diff
@@ -598,6 +600,14 @@ def test_invalid_layouts():
         fig.set_layout_engine("constrained")
 
 
+def test_layout_change_warning():
+    """Raise a warning when an existing layout manager is changed to tight."""
+    for layout in ('constrained', 'compressed'):
```

Comment:
consider using a `parameterize` on the test instead of a loop.

see https://github.com/matplotlib/matplotlib/blob/1133d85bcf55abcdff9ca660da52dc620dab9c0a/lib/matplotlib/tests/test_figure.py#L640 for an example (in the same file)